fundingmanager.go

```diff
@@ -1554,9 +1640,13 @@ func (f *fundingManager) handleFundingLocked(fmsg *fundingLockedMsg) {
 		// that commitment related modifications to this channel can
 		// now proceed.
 		f.barrierMtx.Lock()
-		fndgLog.Tracef("Closing chan barrier for ChanID(%v)", chanID)
-		close(f.newChanBarriers[chanID])
-		delete(f.newChanBarriers, chanID)
+		chanBarrier, ok := f.newChanBarriers[chanID]
```

Comment:
We're missing a conditional select above, on line 1634. This existed _before_ this PR, but we should fix it now while we're in the area:
```go
select {
    case f.cfg.ArbiterChan <- channel:
    case <-f.quit:
        return
}
```

Reply:
Added this, and also a similar select for `peer.newChannels <- newChanMsg` below.